### DISCUSSION TITLE
Shorten version numbers ending in .0

### DISCUSSION THREAD

**Discussion by spawnia** [renovatebot/renovate#24838] [Request Help]
### What would you like Renovate to be able to do?

When adding a new version constraint that specifies a minimum version and ends in `.0`, the version number can be shortened to simply omit `.0`.

For example, see https://github.com/webonyx/graphql-php/pull/1298/files. This pull request adds a new version for `phpunit/phpunit` as `^10.0`. In this case, the version constraint `^10` would be functionally equivalent but shorter and more elegant.

### If you have any ideas on how this should be implemented, please tell us here.

When adding a new version constraint, check if:
- it is a version range (not an exact version number)
- it ends in `.0` (or maybe even `.0.0`)

If so, omit the trailing `.0` to shorten the constraint.

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
1. Is this PHP specific?
2. Is this only a cosmetic change?
3. Why add this complexity if 2=yes?

**Comment by spawnia** [comment]
1. I believe it applies to other package management systems as well. I know that `package.json` also allows shortening `^x.0` or `^x.0.0` to `^x`.
2. Yes, functionally there should be no difference.
3. It might not even make the code more complicated. Apparently there is already some mechanism in place that favors `^10.0` over `^10.0.0`. I believe simplicity and elegance matter.

**Comment by viceice** [comment]
for npm it makes a functionally difference

`^1` allows any `1.x.y` version while `^1.2.0` only allows any `1.2.x` and `^1.2.5` only allows any `1.2.x` which is greater than `1.2.5`

**Comment by viceice** [comment]
Iit also makes a functional difference for composer

https://getcomposer.org/doc/articles/versions.md#next-significant-release-operators

**Comment by spawnia** [comment]
@viceice I am talking specifically about version numbers ending in `.0`. You are correct that `^1.1` is a stronger constraint then `^1`, but `^1.0` is not.

**Comment by rarkins** [comment]
If the user already has `^1.0` and the new version is `^2.0`, you then wouldn't want to simplify to `^2`. As a general rule we wish to retain "precision" and not impose our opinions on others (e.g. ^1 is better than ^1.0 or ^1.0.0 or 1.x).

**Comment by spawnia** [comment]
I can see how upgrading `^1.0` to `^2.0` preserves personal preference. The case is less clear for `^1.1` - I would prefer an upgrade to `^2` there. I certainly would expect `^1` to upgrade to `^2` though.
